includes/compatibility/civicrm.wpml.php

```diff
@@ -100,4 +106,264 @@ public function locale_filter($locale) {
+   *
+   * @param array $languages passed by WPML to modify current path
+   */
+  public function rewrite_civicrm_urls_wpml($languages) {
+
+    // Get the post slug
+    global $post;
+    $post_slug = isset($post->post_name) ? $post->post_name : '';
+    if (empty($post_slug) && isset($post->post_name)) {
+      $post_slug = $post->post_name;
+    }
+
+    // Get CiviCRM basepage slug
+    $civicrm_slug = apply_filters('civicrm_basepage_slug', 'civicrm');
+
+    // Obtain WPML lagnuage negotiation setting
+    $wpml_negotiation = apply_filters('wpml_setting', NULL, 'language_negotiation_type');
+
+    // If this is a CiviCRM Page then let's modify the actual path
+    if ($post_slug == $civicrm_slug) {
+      global $sitepress;
+      $current_url = explode("?", $_SERVER['REQUEST_URI']);
+      $civicrm_url = get_site_url(NULL, $current_url[0]);
+
+      // Remove language from path
+      $wpml_negotiation = apply_filters('wpml_setting', NULL, 'language_negotiation_type');
+      $civicrm_url = $this->wpml_remove_language_from_url($civicrm_url, $sitepress, $wpml_negotiation);
+
+      // Build query string
+      $qs = [];
+      parse_str($_SERVER["QUERY_STRING"], $qs);
+
+      // Strip any WPML languages if they exist
+      unset($qs['lang']);
+      $query = http_build_query($qs);
+
+      // Rebuild CiviCRM links for each language
+      foreach ($languages as &$language) {
+        $url = apply_filters('wpml_permalink', $civicrm_url, $language['language_code']);
+
+        if (!empty($query)) {
+          if ($wpml_negotiation == 3 && strpos($url, '?') !== FALSE) {
+            $language['url'] = $url . '&' . $query;
+          }
+          else {
+            $language['url'] = $url . '?' . $query;
+          }
+        }
+      }
+    }
+
+    return $languages;
+  }
+
+  /**
+   * Remove Language from URL based on WPML configuration
+   *
+   * @since 5.70
+   *
+   * @param $url passed URL to strip language from
+   * @param object $sitepress WPML class
+   * @param int $wpml_negotiation language negotiation setting in WPML
+   *
+   * @return $url base page url without language
+   */
+  private function wpml_remove_language_from_url($url, $sitepress, $wpml_negotiation) {
+    $lang = $this->wpml_get_language_param_for_convert_url($sitepress);
+    if ($lang) {
+      if ($wpml_negotiation == 1) {
+        $url = str_replace('/' . $lang . '/', '/', $url);
+      }
+      elseif ($wpml_negotiation == 3) {
+        $url = str_replace('lang=' . $lang, '', $url);
+      }
+    }
+
+    return $url;
+  }
+
+  /**
+   * Get Language for WPML.
+   * This code was taken from the WPML source, because it couldn't be referenced directly
+   *
+   * @since 5.70
+   *
+   * @param object $sitepress reference to WPML class
+   */
+  private function wpml_get_language_param_for_convert_url($sitepress) {
+    if (!empty($sitepress)) {
+      if (isset($_GET['lang'])) {
+        return filter_var($_GET['lang'], FILTER_SANITIZE_STRING);
```

Comment:
`FILTER_SANITIZE_STRING` is [deprecated in PHP 8.1.0](https://www.php.net/manual/en/filter.filters.sanitize.php). This will write warnings to the logs. [These filters](https://wpml.org/faq/how-to-get-current-language-with-wpml/) might work instead.

Reply:
Thanks for that! I actually found a new [filter](https://wpml.org/faq/how-to-get-current-language-with-wpml/) to grab the current language which is returning the same result. Much more forward thinking then re-using their code which was annoying. I changed the code and it is still working for that.